File: 945. Minimum Increment to Make Array Unique/minIncrementForUnique.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const minIncrementForUnique = (nums) => {
  nums.sort((a, b) => a - b);
  let res = 0;

  for (let i = 1; i < nums.length; i++) {
    if (nums[i - 1] >= nums[i]) {
        const delta = nums[i - 1] - nums[i] + 1;
        nums[i] += delta;
        res += delta;
    }
  }

  return res;
};


const nums = [3, 2, 1, 2, 1, 7];
console.log('minIncrementForUnique: ', minIncrementForUnique(nums));

File: 945. Minimum Increment to Make Array Unique/minIncrementForUnique.py
from typing import List

class Solution:
  def minIncrementForUnique(self, nums: List[int]) -> int:
    nums.sort()
    res = 0
    
    for i in range(1, len(nums)):
      if nums[i - 1] >= nums[i]:
        delta = nums[i - 1] - nums[i] + 1
        nums[i] += delta
        res += delta
        
    return res
    

nums = [3, 2, 1, 2, 1, 7]
solution = Solution()
print('minIncrementForUnique: ', solution.minIncrementForUnique(nums))
